### PR TITLE
Source tests

### PR DESCRIPTION
**Pull request by architekton**
Change the custom flag to conditional building in order to make it work from the root of the repo. Testing commands are as follows: 
* (requires network) `go test -v -tags="dns datasources" ./...`
* (just unit) `go test -v ./...`

### CODE REVIEW

**Comment by fork-while-fork** [comment]
Wow, that is a lot of tests! Is there a reason why we're targeting google so much? Is it because the sources don't have data for the owasp-amass.com domain?

I have a feeling these tests are going to be much harder to maintain if they target a domain we can't control.

File: amass/sources/threatcrowd_test.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import (
+	"log"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/OWASP/Amass/amass/core"
+)
+
+func TestThreaTCrowd(t *testing.T) {
```
Comment (caffix) [inline amass/sources/threatcrowd_test.go]: Typo: Function name should TestThreatCrowd

**Comment by architekton** [comment]
@fork-while-fork I don't mind diversifying across other popular domains. Yes not enough data exists for owasp-amass.com, I tried running `amass -passive -src -d owasp-amass.com` before selecting other domains, but there wasn't much except for cert (crtsh uses owasp-amass.com anyway). Also I was originally going to have one test file. It was going to iterate through an array of sources and just print the error in a `source_name: error` type format but it was requested to have separate tests per source.

File: amass/sources/threatcrowd_test.go (hunk 1 above)
Comment (architekton) [reply amass/sources/threatcrowd_test.go]: Thanks

**Pull request merged**